pkg/controller/provider/web/client.go

```diff
@@ -57,52 +56,92 @@ func NewClient(provider *api.Provider) (client Client, err error) {
 //
 // Provider API client.
 type ProviderClient struct {
-	base.Client
+	restClient base.RestClient
 	// The provider.
 	provider *api.Provider
+	// Finder.
+	finder Finder
 	// Ready.
 	found bool
 }
 
 //
 // Get a resource.
-// Raises ProviderNotReadyErr on 404 and 206
-// when the provider is not yet in the inventory.
-func (r *ProviderClient) Get(resource interface{}, id string) (status int, err error) {
```

Comment:
The `ProviderClient` no longer returns status but specific error types instead to improve usability at this layer.